src/test/java/org/openjdk/shenandoah/CounterTest.java

```diff
@@ -0,0 +1,188 @@
+ *
+ *     Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ *     or visit www.oracle.com if you need additional information or have any
+ *     questions.
+ * ====
+ *
+ * Copyright (c) 2020, Red Hat, Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.shenandoah;
+
+import org.HdrHistogram.Histogram;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.openjdk.shenandoah.RegionAffiliation.OLD;
+import static org.openjdk.shenandoah.RegionAffiliation.YOUNG;
+import static org.openjdk.shenandoah.RegionState.*;
+
+public class CounterTest {
+    List<RegionStat> stats_1 = new ArrayList<>();
+    List<RegionStat> stats_2 = new ArrayList<>();
+    List<RegionStat> stats_3 = new ArrayList<>();
+    List<RegionStat> stats_4 = new ArrayList<>();
+    List<RegionStat> stats_5 = new ArrayList<>();
+    List<RegionStat> stats_6 = new ArrayList<>();
+    List<RegionStat> stats_7 = new ArrayList<>();
+    List<RegionStat> stats_8 = new ArrayList<>();
+    List<RegionStat> stats_9 = new ArrayList<>();
+    List<RegionStat> stats_10 = new ArrayList<>();
+    List<RegionStat> stats_11 = new ArrayList<>();
+    List<RegionStat> stats_12 = new ArrayList<>();
+    List<RegionStat> stats_13 = new ArrayList<>();
+    Snapshot snapshot_1;
+    Snapshot snapshot_2;
+    Snapshot snapshot_3;
+    Snapshot snapshot_4;
+    Snapshot snapshot_5;
+    Snapshot snapshot_6;
+    Snapshot snapshot_7;
+    Snapshot snapshot_8;
+    Snapshot snapshot_9;
+    Snapshot snapshot_10;
+    Snapshot snapshot_11;
+    Snapshot snapshot_12;
+    Snapshot snapshot_13;
+    @Before
+    public void setup() {
+        for (int i = 0; i < 10; i++) {
+            stats_1.add(new RegionStat(EMPTY_UNCOMMITTED, 0));
```

Comment:
I think main reason I use `stats_1` is because it not only tests `emptyUncommitted` but also tests `age0Count`. I am not quite sure how I can name it in this case.